pkg/collector/corechecks/servicediscovery/module/comm_test.go

```diff
@@ -124,9 +133,78 @@ func TestShouldIgnoreComm(t *testing.T) {
 			require.EventuallyWithT(t, func(collect *assert.CollectT) {
 				proc, err := customNewProcess(int32(cmd.Process.Pid))
 				require.NoError(t, err)
-				ignore := shouldIgnoreComm(proc)
+
+				ignore := discovery.shouldIgnoreComm(proc)
 				require.Equal(t, test.ignore, ignore)
-			}, 30*time.Second, 100*time.Millisecond)
+			}, 10*time.Second, 100*time.Millisecond)
+		})
+	}
+}
+
+type ignoredCommTestAttr struct {
+	name  string   // The name of the test.
+	comms []string // list of command names to test
+}
+
+func (a *ignoredCommTestAttr) buildIgnoredCommandsString() string {
+	var builder strings.Builder
+	for _, cmd := range a.comms {
+		builder.WriteString(cmd)
+		builder.WriteString("   ") // intentionally multiple spaces for sensitivity testing
+	}
+	return builder.String()
+}
+
+func TestLoadIgnoredComm(t *testing.T) {
+	tests := []ignoredCommTestAttr{
+		{
+			name:  "empty list of commands",
+			comms: []string{},
+		},
+		{
+			name: "short commands in config list",
+			comms: []string{
+				"cron",
+				"polkitd",
+				"rsyslogd",
+				"bash",
+				"sshd",
+			},
+		},
+		{
+			name: "long commands in config list",
+			comms: []string{
+				"containerd-shim-runc-v2",
+				"calico-node",
+				"unattended-upgrade-shutdown",
+				"bash",
+				"kube-controller-manager",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mockSystemProbe := mock.NewSystemProbe(t)
+			require.NotEmpty(t, mockSystemProbe)
+
+			commsStr := test.buildIgnoredCommandsString()
+			mockSystemProbe.SetWithoutSource("discovery.ignored_command_names", commsStr)
+
+			discovery := newDiscovery()
+			require.NotEmpty(t, discovery)
+
+			require.Equal(t, len(discovery.config.ignoreComms), len(test.comms))
+
+			for _, cmd := range test.comms {
+				if len(cmd) > maxCommLen {
+					_, found := discovery.config.ignoreComms[cmd[:maxCommLen]]
+					assert.True(t, found)
+				} else {
+					_, found := discovery.config.ignoreComms[cmd]
+					assert.True(t, found)
+				}
+			}
```

Comment:
in both cases you expect the same thing. Simplify the check here

Reply:
changed in recent commit